gnucash/gtkbuilder/window-autoclear.glade

```diff
@@ -72,7 +72,9 @@
                 <property name="visible">True</property>
                 <property name="can_focus">False</property>
                 <property name="label" translatable="yes">&lt;b&gt;Auto-Clear Information&lt;/b&gt;
-Use this dialog if you want GnuCash to automatically find which transactions are cleared, given an ending balance. For example, said ending balance can be the current balance given by your online bank.</property>
+Use this dialog if you want GnuCash to automatically find which transactions are cleared, given an ending balance. For example, said ending balance can be the current balance given by your online bank.
+
+&lt;b&gt;Note&lt;/b&gt;: The complexity of this algorithm is quite high. Use this only on a dozen of uncleared splits.</property>
```

Comment:
I had already commented on the old PR, but am happy to write an updated version, as I found the dialog now in the menu.
This would lead to a po entry
```
#: gnucash/gtkbuilder/window-autoclear.glade:74
msgid ""
"<b>Auto-Clear Information</b>\n"
"Use this dialog if you want GnuCash to automatically find which transactions "
"are cleared, given an ending balance. For example, said ending balance can "
"be the current balance given by your online bank.\n"
"\n"
"<b>Note</b>: The complexity of this algorithm is quite high. Use this only "
"on a dozen of uncleared splits."
msgstr ""

```
Invalidating the old entry
```
msgid "<b>Auto-Clear Information</b>\n"
msgstr ""

```
which was already translated into several languages.

I think I would split it in 4 labels:
```
            <child>
              <object class="GtkLabel" id="label1">
                <property name="visible">True</property>
                <property name="can_focus">False</property>
                <property name="label" translatable="yes">&lt;b&gt;About Auto-Clear&lt;/b&gt;</property>
                <property name="use_markup">True</property>
              </object>
              <packing>
                <property name="expand">False</property>
                <property name="fill">False</property>
                <property name="position">0</property>
              </packing>
            </child>
            <child>
              <object class="GtkLabel" id="label2">
                <property name="visible">True</property>
                <property name="can_focus">False</property>
                <property name="label" translatable="yes">Use this dialog if you want GnuCash to automatically find which transactions are cleared, given an ending balance. For example, said ending balance can be the current balance given by your bank online.</property>
                <property name="wrap">True</property>
                <property name="max_width_chars">80</property>
              </object>
              <packing>
                <property name="expand">False</property>
                <property name="fill">False</property>
                <property name="position">1</property>
              </packing>
            </child>
            <child>
              <object class="GtkLabel" id="label3">
                <property name="visible">True</property>
                <property name="can_focus">False</property>
                <property name="label" translatable="yes">&lt;b&gt;Caution!&lt;/b&gt;</property>
                <property name="use_markup">True</property>
              </object>
              <packing>
                <property name="expand">False</property>
                <property name="fill">False</property>
                <property name="position">2</property>
              </packing>
            </child>
            <child>
              <object class="GtkLabel" id="label4">
                <property name="visible">True</property>
                <property name="can_focus">False</property>
                <property name="label" translatable="yes">The complexity of this algorithm is quite high. Use this only on a dozen of uncleared splits.</property>
                <property name="wrap">True</property>
                <property name="max_width_chars">80</property>
              </object>
              <packing>
                <property name="expand">False</property>
                <property name="fill">False</property>
                <property name="position">3</property>
              </packing>
            </child>
            <child>
              <object class="GtkGrid" id="table1">
                <property name="visible">True</property>
                <property name="can_focus">False</property>
                <property name="border_width">6</property>
                <property name="row_spacing">6</property>
                <property name="column_spacing">6</property>
                <child>
                  <object class="GtkLabel" id="end_label">
                    <property name="visible">True</property>
                    <property name="can_focus">False</property>
                    <property name="halign">end</property>
                    <property name="label" translatable="yes">_Ending Balance</property>
                    <property name="use_underline">True</property>
                  </object>
                  <packing>
                    <property name="left_attach">0</property>
                    <property name="top_attach">0</property>
                  </packing>
                </child>
                <child>
                  <object class="GtkBox" id="end_value_box">
                    <property name="visible">True</property>
                    <property name="can_focus">False</property>
                    <property name="hexpand">True</property>
                    <child>
                      <placeholder/>
                    </child>
                  </object>
                  <packing>
                    <property name="left_attach">1</property>
                    <property name="top_attach">0</property>
                  </packing>
                </child>
                <child>
                  <object class="GtkLabel" id="status_label">
                    <property name="visible">True</property>
                    <property name="can_focus">False</property>
                  </object>
                  <packing>
                    <property name="left_attach">0</property>
                    <property name="top_attach">1</property>
                    <property name="width">2</property>
                  </packing>
                </child>
              </object>
              <packing>
                <property name="expand">False</property>
                <property name="fill">True</property>
                <property name="position">4</property>
              </packing>
            </child>
          </object>
          <packing>
            <property name="expand">False</property>
            <property name="fill">False</property>
            <property name="padding">5</property>
            <property name="position">5</property>
          </packing>
        </child>

```
This way we get 2 sections with their own headers.
Textual changes :
1. "About …" seems more appropriate to me than "… Information. (Yes, now invalidate the existing translations, too)
1. "Note" seems too weak, if the user can lock the program for a long time

@Bob-IT any tricks to improve it like avoiding the markup?

Translators would then get:
```
#: gnucash/gtkbuilder/window-autoclear.glade:74
msgid "<b>About Auto-Clear</b>"
msgstr ""

#: gnucash/gtkbuilder/window-autoclear.glade:87
msgid ""
"Use this dialog if you want GnuCash to automatically find which transactions "
"are cleared, given an ending balance. For example, said ending balance can "
"be the current balance given by your bank online."
msgstr ""

#: gnucash/gtkbuilder/window-autoclear.glade:101
msgid "<b>Caution!</b>"
msgstr ""

#: gnucash/gtkbuilder/window-autoclear.glade:114
msgid ""
"The complexity of this algorithm is quite high. Use this only on a dozen of "
"uncleared splits."
msgstr ""

```
Because the chunks are smaller, later changes of the text are easier fixed by the translators.

Reply:
@fellen Yes there is, remove the bold markup and under the 'glade application label common tab' add the style class...
 gnc-class-emphasize-label

This is defined in gnucash-fallback.css as...
.gnc-class-emphasize-label {
  font-size: large;
  font-weight: bold;
}

You could even add gnc-class-highlight but that may be too much.

Reply:
![image](https://user-images.githubusercontent.com/6572538/97617262-d7184d80-1a1d-11eb-921d-8e1d90aa8efe.png)
with
```
                <property name="label" translatable="yes">About Auto-Clear</property>
                <style>
                  <class name="gnc-class-highlight"/>
                </style>
```
and
```
                <property name="label" translatable="yes">Caution!</property>
                <style>
                  <class name="gnc-class-emphasize-label"/>
                </style>
```

Reply:
> @fellen Yes there is

When the wiki is back, we should add it to [GUI Guidelines](https://wiki.gnucash.org/wiki/GUI_Guidelines) or an additional Glade page.

Reply:
I have split off the GUI changes  as PR #811 .
@cristiklein you can remove commit c8bbf83 here.

Reply:
Done, thanks!

Reply:
FWIW the maint branch already has a safeguard (see `#define MAXIMUM_SACK_SIZE 1000000`) whereby the algorithm is aborted when the number of reachable_amounts reaches 1million. This can happen with 20 splits if the amounts are powers of 2, e.g. $1.00 $2.00 $4.00 $8.00 etc, but can easily handle many, many more splits if the amounts can create duplicates e.g. $100.00 $2.00 $2.00 $2.00 $2.00 $2.00 etc. So, IMHO the caution does not need to be too specific; it could state "Caution! The complexity of this algorithm is quite high; it may abort if the search size becomes excessive."